express-notes-api/index.js

```diff
@@ -0,0 +1,129 @@
+const express = require('express');
+const app = express();
+
+app.use('/api/notes', express.json());
+
+const fs = require('fs');
+
+const data = require('./data.json');
+
+app.get('/api/notes', (req, res) => {
+  const finalNotes = [];
+  for (const key in data.notes) {
+    finalNotes.push(data.notes[key]);
+  }
+  res.json(finalNotes);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const notesNumber = req.params.id;
+  if (Number(notesNumber) <= 0 || !Number.isInteger(Number(notesNumber))) {
+    res.status(400).json({
+      error: 'id must be a positive interger'
+    });
+  } else if (!data.notes[notesNumber]) {
+    res.status(404).json({
+      error: `cannot find note with id ${notesNumber}.`
+    });
+  } else {
+    res.status(200).json(data.notes[notesNumber]);
```

Comment:
If you are using status `200` you don't need to explicitly set it, `200` is the default status code.
```JAVASCRIPT
res.json(data.notes[notesNumber]);
```